src/svd/path.py

```diff
@@ -0,0 +1,298 @@
+    """
+
+    __slots__ = "_parts"
+
+    def __init__(self, *parts: Union[str, Sequence[str]]) -> None:
+        split_parts: List[str] = []
+
+        for part in parts:
+            if isinstance(part, str):
+                split_parts.extend(part.split("."))
+            elif isinstance(part, FEPath):
+                split_parts.extend(part)
+            else:
+                sub_parts = (p.split(".") for p in part)
+                split_parts.extend(chain.from_iterable(sub_parts))
+
+        if not split_parts:
+            raise ValueError(f"Empty {self.__class__.__name__} not allowed")
+
+        if any(not p for p in split_parts):
+            raise ValueError(f"Invalid {self.__class__.__name__} parts: {parts}")
+
+        self._parts: Tuple[str, ...] = tuple(split_parts)
+
+    @property
+    def parts(self) -> Tuple[str, ...]:
+        return self._parts
+
+    @property
+    def name(self) -> str:
+        return self.parts[-1]
+
+    @property
+    def parent(self) -> Optional[FEPath]:
+        if len(self.parts) <= 1:
+            return None
+        return FEPath(*self.parts[:-1])
+
+    @property
+    def stem(self) -> str:
+        return self.name
+
+    def to_xpath(self) -> str:
+        """:return: An XPath expression that can be used to locate XML elements having this path"""
+        return "." + "".join((f"/*[name='{p}']" for p in self.parts))
+
+    def __repr__(self) -> str:
+        return ".".join(self.parts)
+
+
+class EPath(AbstractEPath[Union[str, int]]):
+    """
+    Path to a SVD element.
+    """
+
+    __slots__ = "_parts"
+
+    def __init__(self, *parts: Union[str, int, Sequence[Union[str, int]]]) -> None:
+        processed_parts = self._process_parts(parts)
+
+        if not processed_parts:
+            raise ValueError(f"Empty {self.__class__.__name__} not allowed")
+
+        self._parts: Tuple[Union[str, int], ...] = tuple(processed_parts)
+
+    @property
+    def parts(self) -> Tuple[Union[str, int], ...]:
+        return self._parts
+
+    @property
+    def name(self) -> str:
+        for i in reversed(range(len(self._parts))):
+            if isinstance(self._parts[i], str):
+                return self._format_parts(self._parts[i:])
+        assert False
+
+    @property
+    def stem(self) -> str:
+        for part in reversed(self._parts):
+            if isinstance(part, str):
+                return part
+        assert False
+
+    @property
+    def parent(self) -> Optional[EPath]:
+        if len(self._parts) <= 1:
+            return None
+        return EPath(*self._parts[:-1])
+
+    @property
+    def element_index(self) -> Optional[int]:
+        """Index of the register in the parent array, if applicable."""
+        if not isinstance(self[-1], int):
+            return None
+        return self[-1]
+
+    def to_flat(self) -> FEPath:
+        """Convert the regular path to the equivalent flat path."""
+        return FEPath(*(p for p in self.parts if not isinstance(p, int)))
```

Comment:
`self.parts` won't change after the class instance gets created, right? If so then maybe this can be a `cachedproperty` to save some processing?